File: DatabaseBackend/PasswordHasher.java
class PasswordHasher {

    private static int hash = 987875;
    private static int prime1 = 488353;
    private static int prime2 = 7687;


    PasswordHasher () {

    }

    public static String hashPassword(String password) {

        int passwordLength = password.length();
        int charInt = 1;
        for (int i = 0; i < passwordLength; i++) {
            charInt = charInt * password.charAt(i);
        }

        int hash1 = hash * charInt*prime2 + prime1/passwordLength;
        int hash2 = prime2/passwordLength + prime1*charInt;
        String hashConc = Integer.toString(hash1).substring(1) + Integer.toString(hash2).substring(1);

        return hashConc;
    }

}

File: gui/src/PasswordHasher.js

class PasswordHasher {

    constructor () {}

    hashPassword (password) {

        let hash = 9;
        let prime1 = 488353;
        let prime2 = 7687;

        let passwordLength = password.length;
        let charInt = 1;

        var i;
        for (i = 0; i < passwordLength; i++) {
            charInt = charInt * password.charCodeAt(i);
        }

        let hash1 = hash * charInt * prime2 + prime1 / passwordLength;
        let hash2 = prime2 / passwordLength + prime1 * charInt;
        let hashConc = hash1.toString().substring(1, hash1.length) + hash2.toString().substring(1, hash2.length);

        return hashConc;
    }
}

export default PasswordHasher;